sigal/video.py

```diff
@@ -78,30 +78,36 @@ def video_size(source, converter='ffmpeg'):
     return x, y
 
 
-def generate_video(source, outname, settings, options=None):
-    """Video processor.
+def get_dimensions(source, converter, output_size):
+    """Figure out src and dest width and height for video.
 
     :param source: path to a video
     :param outname: path to the generated video
     :param settings: settings dict
-    :param options: array of options passed to ffmpeg
 
     """
     logger = logging.getLogger(__name__)
 
     # Don't transcode if source is in the required format and
     # has fitting datedimensions, copy instead.
-    converter = settings['video_converter']
     w_src, h_src = video_size(source, converter=converter)
-    w_dst, h_dst = settings['video_size']
+    w_dst, h_dst = output_size
     logger.debug('Video size: %i, %i -> %i, %i', w_src, h_src, w_dst, h_dst)
+    return {'src': (w_src, h_src), 'dst': (w_dst, h_dst)}
```

Comment:
Why not putting this directly in `get_resize_options` to avoid the dict packing/unpacking ? I think it's the only place where it is used.